### ISSUE TITLE
Gateway Discovery: implement marking Gateway proxy service ready after config has been pushed

### ISSUE THREAD

**Issue by pmalek** [Kong/kubernetes-ingress-controller#3499]
#3421 introduces service discovery based on provided Service namespaced name.

This will make it so that a new Gateway can be made a part of proxy Service before an initial config push to it is made.

This does retain the current behavior but would be nice to have this improvement introduced.

Details of the implementation are described in the [design doc](https://docs.google.com/document/d/1FdTBPD-8PACD93TjSzrBG_CSeMHir9CrVAoWFM5ru_4/edit?usp=sharing).

### Acceptance criteria

- [ ] As a User I can use config readiness endpoint to check via status code of the response if config has been applied
- [ ] As a User I can rely on config readiness endpoint to be used in deployed manifests - all in one manifests - to ensure that only the Pods with Kong Gateways that already received config can be taken into account when routing traffic (via kubernetes readiness probes)

**Comment by pmalek** [comment]
This is blocked on implementation of FTI-3081.

**Comment by pmalek** [comment]
@mheap Added!

**Comment by mflendrich** [comment]
xposting @pmalek's comment from #4064 

> This is actually an interesting issue because:
> 
> - as mentioned in the description, we'll have to reassess whether to consider not-ready (in k8s meaning) Pods are ready to receive configuration, otherwise they will never get ready
> - what do we want to do with most of the demos that are performed with KIC where a user installs KIC and deploys Gateways and proves that it works by showing that `/` doesn't return a 404
